spec/services/tslr/school_eligibility_spec.rb

```diff
@@ -0,0 +1,93 @@
+          it { is_expected.to be false }
+        end
+      end
+
+      context "and it is a free school" do
+        let(:free_school_attributes) { {school_type_group: :free_schools} }
+
+        context "and it has an education phase of secondary or middle deemed secondary" do
+          let(:school_attributes) { free_school_attributes.merge({phase: :secondary}) }
+          it { is_expected.to be true }
+        end
+
+        context "and it has an education phase of nursery, primary, middle deemed primary, 16+ or all through" do
+          let(:school_attributes) { free_school_attributes.merge({phase: :middle_deemed_primary}) }
+          it { is_expected.to be false }
+        end
+      end
+
+      context "and it is a college" do
+        let(:college_attributes) { {school_type_group: :colleges} }
+
+        context "and it has an education phase of secondary or middle deemed secondary" do
+          let(:school_attributes) { college_attributes.merge({phase: :middle_deemed_secondary}) }
+          it { is_expected.to be true }
+        end
+
+        context "and it has an education phase of nursery, primary, middle deemed primary, 16+ or all through" do
+          let(:school_attributes) { college_attributes.merge({phase: :sixteen_plus}) }
+          it { is_expected.to be false }
+        end
+      end
+
+      context "and it is a LA maintained school" do
+        let(:la_maintained_attributes) { {school_type_group: :la_maintained} }
+
+        context "and it has an education phase of secondary or middle deemed secondary" do
+          let(:school_attributes) { la_maintained_attributes.merge({phase: :secondary}) }
+          it { is_expected.to be true }
+        end
+
+        context "and it has an education phase of nursery, primary, middle deemed primary, 16+ or all through" do
+          let(:school_attributes) { la_maintained_attributes.merge({phase: :nursery}) }
+          it { is_expected.to be false }
+        end
+      end
+
+      context "and it is a special school" do
+        let(:special_school_attributes) { {school_type_group: :special_schools} }
+
+        context "and it is a state funded special school" do
+          let(:school_attributes) { special_school_attributes.merge({school_type: :community_school})}
+          it { is_expected.to be true }
+        end
+
+        context "and it is an independent special school" do
+          let(:school_attributes) { special_school_attributes.merge({school_type: :other_independent_special_school})}
+          it { is_expected.to be false }
+        end
+      end
+
+      context "and it is not a state funded school" do
+        let(:school_attributes) { {school_type_group: :independent_schools} }
+        it { is_expected.to be false }
+      end
+    end
+
+    context "when it is not in an eligible area" do
+      let(:local_authority) { build(:local_authority, :ineligible) }
+      let(:school_attributes) { {} }
```

Comment:
For this spec to be a bit more legit, I think the school attributes should be for an otherwise eligible school, i.e. state funded in an eligible phase. Otherwise it might be failing eligibility for any one of the other checks.

Reply:
Ah, sorry, just noticed these specs are using the factory that already has eligible attributes, so that isn't necessary.